### ISSUE TITLE
Fresh Plastid install is incompatible with pysam version

### ISSUE THREAD

**Issue by lparsons** [bioconda/bioconda-utils#566]
It seems that the version of pysam isn't pinned when installing plastid. Since pysam isn't explicitly on this list: https://github.com/bioconda/bioconda-utils/blob/master/bioconda_utils/bioconda_utils-conda_build_config.yaml I'm guessing it won't get pinned automatically. I can think of two ways to handle this:
1. Add pysam to the list of automatically pinned packages
2. Add htslib directly to the dependency list of plastid (and really any other package that lists pysam).

Or perhaps I'm totally off. @bioconda/core any thoughts?

**Comment by dpryan79** [comment]
Is plastid compiling against pysam?!?

**Comment by lparsons** [comment]
Yes, see https://github.com/bioconda/bioconda-recipes/issues/12494

**Comment by dpryan79** [comment]
That's the only package I've ever seen that compiles (in this case seemingly using Cython) against pysam. In this case `pin_compatible()` should suffice.

**Comment by lparsons** [comment]
@dpryan79 Thanks, I'll do that. Any tips on how I should best specify that? Can I specify a min version? Tips on setting `min_pin` and `max_pin`? I'm not exactly clear on how those work and didn't see any docs.

**Comment by lparsons** [comment]
Also, should I pin numpy? Seems like it, but I'm not sure how to specify a min/max version then...

**Comment by dpryan79** [comment]
I think the `min_pin` can be left as is and the `max_pin` set to `x.x`. When you make the PR, download the tarball that the bot posts and look in `info/recipe/meta.yaml` at the pysam range specified under `run:`. Make sure that only specifies a minor version range (e.g., `>=1.8.0,<1.9`). The reality, of course, is that simply bumping the build number will probably get things working for about a year :P

**Comment by lparsons** [comment]
Oops. Sorry, I see. Specify the version in `build` and the `pin_compatible` in `run`. Too late on a Friday.. ;-)

**Comment by epruesse** [comment]
I wrote something compiling against pysam a while ago. Yes - would be cython, always.

**Comment by epruesse** [comment]
xref https://github.com/bioconda/bioconda-recipes/issues/12494

**Comment by epruesse** [comment]
Can you add a PR against PR and close this?
